ipatests/test_integration/test_trust.py

```diff
@@ -501,8 +501,24 @@ def test_user_gid_uid_resolution_in_forest_trust_with_shared_secret(self):
                          % (re.escape(self.ad_domain),
                             re.escape(self.ad_domain))
 
-        assert re.search(
-            testuser_regex, result.stdout_text), result.stdout_text
+        # SSSD versions < 2.0.0 contain a bug
+        # https://pagure.io/SSSD/sssd/issue/3911
+        # In other tests it is mitigated by 60 second sleep time in
+        # tasks.clear_sssd_cache. But this particular setup requires more
+        # time to start successfully fetch user from AD domain.
+        timeout = 300
+        start = time.time()
+        while True:
+            result = self.master.run_command(['getent', 'passwd', testuser],
+                                             ok_returncode=[0, 2])
+            try:
+                assert result.returncode == 0
+                assert re.search(
+                    testuser_regex, result.stdout_text), result.stdout_text
+                break
+            except AssertionError:
+                if time.time() - start > timeout:
+                    raise
```

Comment:
You can add a sleep(1) if we didn't reach the timeout yet.

Reply:
oops... missed the most important part